frontend/catalyst/jax_primitives.py

```diff
@@ -528,16 +529,28 @@ def _apply_registered_pass_lowering(
         """
 
     if first_op_in_block.name == "transform.apply_registered_pass":
+        """
```

Comment:
This reverse ordering of jaxpr primitives and mlir aaply_registered_pass is only due to the fact that we are repeatedly wrapping.

If we exclude ZNE, then all passes will be of the same format (https://github.com/PennyLaneAI/catalyst/blob/main/mlir/include/Quantum/Transforms/Passes.td#L94), and the wrapping is unnecessary since we can just bind the apply_registered_pass primitive in the pipeline decorator, instead of having it call APIs.